src/app/json-editor/json-editor.component.ts

```diff
@@ -44,25 +55,22 @@ export class JsonEditorComponent implements OnInit {
       if (this.jsonSource) {
         this.jsonTarget = JSON.stringify(JSON.parse(this.jsonSource));
       }
-      alert('JSON is not well formated.');
+      else {
+        this.handleError();
+      }
     } catch (e) {
-      this.handleError(e);
+      this.handleError();
     }
   }
 
   beautifyJSON() {
     try {
       this.jsonTarget = JSON.stringify(JSON.parse(this.jsonSource), null, '    ');
     } catch (e) {
-      this.handleError(e);
+      this.handleError();
```

Comment:
It looks you are not using actual error message, it's showing a static message every time on the modal popup. that should be dynamic

Reply:
Resolved it.